src/VisualStudio/CSharp/Impl/Debugging/CSharpBreakpointResolutionService.cs

```diff
@@ -15,29 +18,33 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.Debugging
     [ExportLanguageService(typeof(IBreakpointResolutionService), LanguageNames.CSharp), Shared]
     internal partial class CSharpBreakpointResolutionService : IBreakpointResolutionService
     {
-        internal static async Task<BreakpointResolutionResult> GetBreakpointAsync(Document document, int position, CancellationToken cancellationToken)
+        /// <summary>
+        /// Returns null if a breakpoint can't be placed at the specified position.
+        /// </summary>
+        public async Task<BreakpointResolutionResult> ResolveBreakpointAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken)
         {
-            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            if (!BreakpointSpans.TryGetBreakpointSpan(tree, position, cancellationToken, out var span))
+            try
             {
-                return null;
-            }
+                var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                if (!BreakpointSpans.TryGetBreakpointSpan(tree, textSpan.Start, cancellationToken, out var span))
+                {
+                    return null;
+                }
+
+                if (span.Length == 0)
+                {
+                    return BreakpointResolutionResult.CreateLineResult(document);
+                }
 
-            if (span.Length == 0)
+                return BreakpointResolutionResult.CreateSpanResult(document, span);
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
```

Comment:
Because this method is async method and async methods swallow exceptions, so the stack is gone by the time the exception gets rethrown in the caller.